File: part_1/hw4/ex_A_3_3_h.m
cvx_begin
    variable x;
    variable y;
    variable z;
    minimize(norm(x + y + z + 1));
    subject to
        x + z <= 1 + geo_mean([x, (y - quad_over_lin(z, x))])
        x >= 0;
        y >= 0;
cvx_end

fprintf('status:'); 
disp(cvx_status);

fprintf('optimal value:'); 
disp(cvx_optval )

fprintf('optimal x:\n'); 
disp(x)

fprintf('optimal y:\n'); 
disp(y)

fprintf('optimal z:\n'); 
disp(z)

File: part_1/hw4/ex_A_3_3_h.py
import cvxpy as cp

# Create two scalar optimization variables.
x = cp.Variable()
y = cp.Variable()
z = cp.Variable()

# Create two constraints.
constraints = [x + z <= 1 + cp.geo_mean(x, y - cp.quad_over_lin(z, z)),
               x >= 0, y >= 0]

# Form objective.
obj = cp.Minimize((x - y + 2)**2)

# Form and solve problem.
prob = cp.Problem(obj, constraints)
prob.solve()  # Returns the optimal value.
print("status:", prob.status)
print("optimal value", prob.value)
print("optimal var", x.value, y.value)

